src/histbuf.rs

```diff
@@ -70,10 +70,26 @@ impl<T, const N: usize> HistoryBuffer<T, N> {
         }
     }
 
-    /// Clears the buffer, replacing every element with the default value of
-    /// type `T`.
+    /// Clear the buffer, removing all entries.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use heapless::HistoryBuffer;
+    ///
+    /// let mut buf: HistoryBuffer<u8, 16> = HistoryBuffer::new();
+    ///
+    /// buf.write(3);
+    /// buf.write(5);
+    /// buf.extend(&[4, 4]);
+    /// assert!(!buf.is_empty());
+    ///
+    /// buf.clear();
+    /// assert!(buf.is_empty());
+    /// ```
     pub fn clear(&mut self) {
-        *self = Self::new();
+        self.write_at = 0;
+        self.filled = false;
```

Comment:
this should drop all the items that were previously in the container. (`*self = ...` did it implicitly because it dropped self before the assignment)

Reply:
:man_facepalming: thanks!  I added a test for this separately here: #435